src/main/java/net/dries007/tfc/util/PowderKegExplosion.java

```diff
@@ -0,0 +1,147 @@
+                double d1 = ((float) blockpos.getY() + this.level.random.nextFloat());
+                double d2 = ((float) blockpos.getZ() + this.level.random.nextFloat());
+                double d3 = d0 - this.x;
+                double d4 = d1 - this.y;
+                double d5 = d2 - this.z;
+                double d6 = Mth.sqrt((float) (d3 * d3 + d4 * d4 + d5 * d5));
+                d3 = d3 / d6;
+                d4 = d4 / d6;
+                d5 = d5 / d6;
+                double d7 = 0.5d / (d6 / (double) this.size + 0.1d);
+                d7 = d7 * (double) (this.level.random.nextFloat() * this.level.random.nextFloat() + 0.3f);
+                d3 = d3 * d7;
+                d4 = d4 * d7;
+                d5 = d5 * d7;
+                level.addParticle(ParticleTypes.EXPLOSION, (d0 + this.x) / 2.0d, (d1 + this.y) / 2.0d, (d2 + this.z) / 2.0d, d3, d4, d5);
+                level.addParticle(ParticleTypes.EXPLOSION, d0, d1, d2, d3, d4, d5);
+            }
+
+            if (!blockstate.isAir())
+            {
+                BlockPos blockpos1 = blockpos.immutable();
+                this.level.getProfiler().push("explosion_blocks");
+                if (blockstate.canDropFromExplosion(this.level, blockpos, this) && this.level instanceof ServerLevel) {
+                    BlockEntity blockentity = blockstate.hasBlockEntity() ? this.level.getBlockEntity(blockpos) : null;
+                    LootContext.Builder lootcontext$builder = (new LootContext.Builder((ServerLevel)this.level)).withRandom(this.level.random).withParameter(LootContextParams.ORIGIN, Vec3.atCenterOf(blockpos)).withParameter(LootContextParams.TOOL, ItemStack.EMPTY).withOptionalParameter(LootContextParams.BLOCK_ENTITY, blockentity).withOptionalParameter(LootContextParams.THIS_ENTITY, this.source);
+
+                    blockstate.getDrops(lootcontext$builder).forEach((drop) -> {
+                        addBlockDrops(allDrops, drop, blockpos1);
+                    });
+                }
+
+                blockstate.onBlockExploded(this.level, blockpos, this);
+                this.level.getProfiler().pop();
+            }
+
+        }
+
+        for(Pair<ItemStack, BlockPos> pair : allDrops) {
+            Block.popResource(this.level, pair.getSecond(), pair.getFirst());
+        }
+    }
+
+    private static void addBlockDrops(ObjectArrayList<Pair<ItemStack, BlockPos>> allDrops, ItemStack drop, BlockPos dropPos) {
+        int i = allDrops.size();
+
+        for(int j = 0; j < i; ++j) {
+            Pair<ItemStack, BlockPos> pair = allDrops.get(j);
+            ItemStack itemstack = pair.getFirst();
+            if (ItemEntity.areMergable(itemstack, drop)) {
+                ItemStack itemstack1 = ItemEntity.merge(itemstack, drop, 16);
+                allDrops.set(j, Pair.of(itemstack1, pair.getSecond()));
+                if (drop.isEmpty()) {
+                    return;
+                }
```

Comment:
this code style needs to be fixed. Import the code style XML from the repository and apply it to all your changes